File: 0965-univalued-binary-tree/0965-univalued-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    boolean isUnivalTree_01(TreeNode node,int val) {
        if(node == null) return true;   
        
        boolean hasLeft = isUnivalTree_01(node.left,val);
        boolean hasRight = isUnivalTree_01(node.right,val);
        
        if(node.val == val && hasLeft && hasRight)return true;
        else return false;
    }
    
    public boolean isUnivalTree(TreeNode root) {
        if(root == null) return true;
        int val = root.val;
        return isUnivalTree_01(root,val);
    }
}

File: 0965-univalued-binary-tree/0965-univalued-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isUnivalTree_01(TreeNode* root, int val) {
        //Base Case
        if(root == nullptr) return true;
        //Left Recursive Call
        bool hasLeftSameVal = isUnivalTree_01(root->left,val);
        //Right Recursive Call
        bool hasRightSameVal = isUnivalTree_01(root->right,val);
        
        if(hasLeftSameVal && hasRightSameVal && root->val == val) return true;
        else return false;
    }
    
    bool isUnivalTree(TreeNode* root) {
        //Base Case
        if(root == nullptr) return true;
        int val = root->val;
        return isUnivalTree_01(root,val);
    }
};